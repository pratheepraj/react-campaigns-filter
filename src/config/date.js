export const DATE_START = 'DATE_START';
export const DATE_END = 'DATE_END';
export const DEFAULT_DATE_FORMAT = 'DD/MM/YYYY';
export const PLACEHOLDER_FORMAT = 'dd/mm/yyyy';

// This should ideally be in a language translation file
export const START_DATE_ERROR = "Start date can't be later than End date";
export const END_DATE_ERROR = "End date can't be earlier than Start date";
