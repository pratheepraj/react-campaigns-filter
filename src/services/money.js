import currency from 'currency.js';

// Outputs in the format 000,000,000
export const formatMoney = value => currency(value, { symbol: '', precision: 0 }).format(true);
