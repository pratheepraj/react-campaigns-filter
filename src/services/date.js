import moment from 'moment';
import { DEFAULT_DATE_FORMAT } from '../config/date';

// Calculates whether a given date is within two other dates
export const inRange = (date, startDate, endDate) => date >= startDate && date <= endDate;

// Converts date from format 'MM/DD/YYYY' to timestamp
export const convertToTimestamp = (date = '') => {
  const split = date.split('/');
  return Date.parse([split[2], split[0], split[1]].join('-'));
};

// Formats date to default format
export const formatDate = date => moment(date).format(DEFAULT_DATE_FORMAT);
