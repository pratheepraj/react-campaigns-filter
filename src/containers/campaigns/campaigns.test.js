import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import Campaigns from './campaigns.jsx';
import { convertToTimestamp } from '../../services/date';

describe('Campaigns', () => {
  let component = null;
  let instance = null;

  beforeEach(() => {
    component = shallow(<Campaigns />);
    instance = component.instance();
  });

  it('renders correctly', () => {
    expect(component).toMatchSnapshot();
  });

  it('convertDatesToTimestamp: Converts dates in string format to timestamp', () => {
    const data = [
      { id: 1, name: 'Divavu', startDate: '07-12-2018', endDate: '07-30-2018', Budget: 88377 },
    ];

    const timestamp = Date.parse(data[0].startDate);
    const converted = instance.convertDatesToTimestamp(data);
    expect(converted[0].startDate).toEqual(timestamp);
  });

  it('removeCampaignsWithBadDate: Removes campaigns with end date prior to the start date', () => {
    const data = [
      {
        id: 1,
        startDate: convertToTimestamp('01/18/2020'),
        endDate: convertToTimestamp('05/18/2020'),
      },
      {
        id: 2,
        startDate: convertToTimestamp('01/18/2018'),
        endDate: convertToTimestamp('05/18/2016'),
      },
    ];

    expect(instance.removeCampaignsWithBadDate(data).length).toEqual(1);
  });

  it('filterByCampaignName', () => {
    const data = [
      { id: 1, name: 'Divavu' },
      { id: 2, name: 'Jaxspan' },
      { id: 3, name: 'Jaxspanish' },
    ];

    const filtered = instance.filterByCampaignName(data, 'jaxs');
    expect(filtered.length).toEqual(2);
    expect(filtered[0].id).toEqual(2);
    expect(filtered[1].id).toEqual(3);
  });

  it('filterByDate', () => {
    const data = [
      {
        id: 1,
        startDate: convertToTimestamp('01/18/2018'),
        endDate: convertToTimestamp('05/18/2018'),
      },
      {
        id: 2,
        startDate: convertToTimestamp('01/18/2019'),
        endDate: convertToTimestamp('05/18/2019'),
      },
      {
        id: 3,
        startDate: convertToTimestamp('01/18/2020'),
        endDate: convertToTimestamp('05/18/2020'),
      },
    ];

    const rangeStartDate = convertToTimestamp('01/01/2020');
    const rangeEndDate = convertToTimestamp('12/01/2020');

    const filtered = instance.filterByDate(data, rangeStartDate, rangeEndDate);
    expect(filtered.length).toEqual(1);
    expect(filtered[0].id).toEqual(3);
  });
});
