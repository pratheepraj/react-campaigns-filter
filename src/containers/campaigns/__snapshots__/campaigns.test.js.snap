// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Campaigns renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Campaigns />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          className="campaigns__header"
>
          <DateRange
                    endDate={1546128000000}
                    onChange={[Function]}
                    onReset={[Function]}
                    startDate={1468281600000}
          />
          <searchBox
                    onChange={[Function]}
          />
</header>,
        <main>
          <CampaignList
                    campaigns={
                              Array [
                                        Object {
                                          "Budget": 837733,
                                          "endDate": 1532908800000,
                                          "id": 1000,
                                          "name": "Ant-Man",
                                          "startDate": 1499817600000,
                                        },
                                        Object {
                                          "Budget": 608715,
                                          "endDate": 1532217600000,
                                          "id": 1001,
                                          "name": "Black Panter",
                                          "startDate": 1531353600000,
                                        },
                                        Object {
                                          "Budget": 239507,
                                          "endDate": 1537574400000,
                                          "id": 1003,
                                          "name": "Gotham City",
                                          "startDate": 1526774400000,
                                        },
                                        Object {
                                          "Budget": 1349838,
                                          "endDate": 1546128000000,
                                          "id": 1004,
                                          "name": "R2-D2",
                                          "startDate": 1508025600000,
                                        },
                                        Object {
                                          "Budget": 884850,
                                          "endDate": 1532217600000,
                                          "id": 1005,
                                          "name": "Star City",
                                          "startDate": 1531353600000,
                                        },
                                        Object {
                                          "Budget": 85313,
                                          "endDate": 1500940800000,
                                          "id": 1006,
                                          "name": "Green Arrow",
                                          "startDate": 1500768000000,
                                        },
                                        Object {
                                          "Budget": 109448,
                                          "endDate": 1532217600000,
                                          "id": 1007,
                                          "name": "Stranger Things",
                                          "startDate": 1531353600000,
                                        },
                                        Object {
                                          "Budget": 255442,
                                          "endDate": 1469145600000,
                                          "id": 1008,
                                          "name": "Klingons",
                                          "startDate": 1468281600000,
                                        },
                                      ]
                    }
          />
</main>,
      ],
      "className": "campaigns",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <DateRange
              endDate={1546128000000}
              onChange={[Function]}
              onReset={[Function]}
              startDate={1468281600000}
/>,
            <searchBox
              onChange={[Function]}
/>,
          ],
          "className": "campaigns__header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "endDate": 1546128000000,
              "onChange": [Function],
              "onReset": [Function],
              "startDate": 1468281600000,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "onChange": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <CampaignList
            campaigns={
                        Array [
                                    Object {
                                      "Budget": 837733,
                                      "endDate": 1532908800000,
                                      "id": 1000,
                                      "name": "Ant-Man",
                                      "startDate": 1499817600000,
                                    },
                                    Object {
                                      "Budget": 608715,
                                      "endDate": 1532217600000,
                                      "id": 1001,
                                      "name": "Black Panter",
                                      "startDate": 1531353600000,
                                    },
                                    Object {
                                      "Budget": 239507,
                                      "endDate": 1537574400000,
                                      "id": 1003,
                                      "name": "Gotham City",
                                      "startDate": 1526774400000,
                                    },
                                    Object {
                                      "Budget": 1349838,
                                      "endDate": 1546128000000,
                                      "id": 1004,
                                      "name": "R2-D2",
                                      "startDate": 1508025600000,
                                    },
                                    Object {
                                      "Budget": 884850,
                                      "endDate": 1532217600000,
                                      "id": 1005,
                                      "name": "Star City",
                                      "startDate": 1531353600000,
                                    },
                                    Object {
                                      "Budget": 85313,
                                      "endDate": 1500940800000,
                                      "id": 1006,
                                      "name": "Green Arrow",
                                      "startDate": 1500768000000,
                                    },
                                    Object {
                                      "Budget": 109448,
                                      "endDate": 1532217600000,
                                      "id": 1007,
                                      "name": "Stranger Things",
                                      "startDate": 1531353600000,
                                    },
                                    Object {
                                      "Budget": 255442,
                                      "endDate": 1469145600000,
                                      "id": 1008,
                                      "name": "Klingons",
                                      "startDate": 1468281600000,
                                    },
                                  ]
            }
/>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "campaigns": Array [
              Object {
                "Budget": 837733,
                "endDate": 1532908800000,
                "id": 1000,
                "name": "Ant-Man",
                "startDate": 1499817600000,
              },
              Object {
                "Budget": 608715,
                "endDate": 1532217600000,
                "id": 1001,
                "name": "Black Panter",
                "startDate": 1531353600000,
              },
              Object {
                "Budget": 239507,
                "endDate": 1537574400000,
                "id": 1003,
                "name": "Gotham City",
                "startDate": 1526774400000,
              },
              Object {
                "Budget": 1349838,
                "endDate": 1546128000000,
                "id": 1004,
                "name": "R2-D2",
                "startDate": 1508025600000,
              },
              Object {
                "Budget": 884850,
                "endDate": 1532217600000,
                "id": 1005,
                "name": "Star City",
                "startDate": 1531353600000,
              },
              Object {
                "Budget": 85313,
                "endDate": 1500940800000,
                "id": 1006,
                "name": "Green Arrow",
                "startDate": 1500768000000,
              },
              Object {
                "Budget": 109448,
                "endDate": 1532217600000,
                "id": 1007,
                "name": "Stranger Things",
                "startDate": 1531353600000,
              },
              Object {
                "Budget": 255442,
                "endDate": 1469145600000,
                "id": 1008,
                "name": "Klingons",
                "startDate": 1468281600000,
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "main",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="campaigns__header"
>
            <DateRange
                        endDate={1546128000000}
                        onChange={[Function]}
                        onReset={[Function]}
                        startDate={1468281600000}
            />
            <searchBox
                        onChange={[Function]}
            />
</header>,
          <main>
            <CampaignList
                        campaigns={
                                    Array [
                                                Object {
                                                  "Budget": 837733,
                                                  "endDate": 1532908800000,
                                                  "id": 1000,
                                                  "name": "Ant-Man",
                                                  "startDate": 1499817600000,
                                                },
                                                Object {
                                                  "Budget": 608715,
                                                  "endDate": 1532217600000,
                                                  "id": 1001,
                                                  "name": "Black Panter",
                                                  "startDate": 1531353600000,
                                                },
                                                Object {
                                                  "Budget": 239507,
                                                  "endDate": 1537574400000,
                                                  "id": 1003,
                                                  "name": "Gotham City",
                                                  "startDate": 1526774400000,
                                                },
                                                Object {
                                                  "Budget": 1349838,
                                                  "endDate": 1546128000000,
                                                  "id": 1004,
                                                  "name": "R2-D2",
                                                  "startDate": 1508025600000,
                                                },
                                                Object {
                                                  "Budget": 884850,
                                                  "endDate": 1532217600000,
                                                  "id": 1005,
                                                  "name": "Star City",
                                                  "startDate": 1531353600000,
                                                },
                                                Object {
                                                  "Budget": 85313,
                                                  "endDate": 1500940800000,
                                                  "id": 1006,
                                                  "name": "Green Arrow",
                                                  "startDate": 1500768000000,
                                                },
                                                Object {
                                                  "Budget": 109448,
                                                  "endDate": 1532217600000,
                                                  "id": 1007,
                                                  "name": "Stranger Things",
                                                  "startDate": 1531353600000,
                                                },
                                                Object {
                                                  "Budget": 255442,
                                                  "endDate": 1469145600000,
                                                  "id": 1008,
                                                  "name": "Klingons",
                                                  "startDate": 1468281600000,
                                                },
                                              ]
                        }
            />
</main>,
        ],
        "className": "campaigns",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <DateRange
                endDate={1546128000000}
                onChange={[Function]}
                onReset={[Function]}
                startDate={1468281600000}
/>,
              <searchBox
                onChange={[Function]}
/>,
            ],
            "className": "campaigns__header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "endDate": 1546128000000,
                "onChange": [Function],
                "onReset": [Function],
                "startDate": 1468281600000,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "onChange": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <CampaignList
              campaigns={
                            Array [
                                          Object {
                                            "Budget": 837733,
                                            "endDate": 1532908800000,
                                            "id": 1000,
                                            "name": "Ant-Man",
                                            "startDate": 1499817600000,
                                          },
                                          Object {
                                            "Budget": 608715,
                                            "endDate": 1532217600000,
                                            "id": 1001,
                                            "name": "Black Panter",
                                            "startDate": 1531353600000,
                                          },
                                          Object {
                                            "Budget": 239507,
                                            "endDate": 1537574400000,
                                            "id": 1003,
                                            "name": "Gotham City",
                                            "startDate": 1526774400000,
                                          },
                                          Object {
                                            "Budget": 1349838,
                                            "endDate": 1546128000000,
                                            "id": 1004,
                                            "name": "R2-D2",
                                            "startDate": 1508025600000,
                                          },
                                          Object {
                                            "Budget": 884850,
                                            "endDate": 1532217600000,
                                            "id": 1005,
                                            "name": "Star City",
                                            "startDate": 1531353600000,
                                          },
                                          Object {
                                            "Budget": 85313,
                                            "endDate": 1500940800000,
                                            "id": 1006,
                                            "name": "Green Arrow",
                                            "startDate": 1500768000000,
                                          },
                                          Object {
                                            "Budget": 109448,
                                            "endDate": 1532217600000,
                                            "id": 1007,
                                            "name": "Stranger Things",
                                            "startDate": 1531353600000,
                                          },
                                          Object {
                                            "Budget": 255442,
                                            "endDate": 1469145600000,
                                            "id": 1008,
                                            "name": "Klingons",
                                            "startDate": 1468281600000,
                                          },
                                        ]
              }
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "campaigns": Array [
                Object {
                  "Budget": 837733,
                  "endDate": 1532908800000,
                  "id": 1000,
                  "name": "Ant-Man",
                  "startDate": 1499817600000,
                },
                Object {
                  "Budget": 608715,
                  "endDate": 1532217600000,
                  "id": 1001,
                  "name": "Black Panter",
                  "startDate": 1531353600000,
                },
                Object {
                  "Budget": 239507,
                  "endDate": 1537574400000,
                  "id": 1003,
                  "name": "Gotham City",
                  "startDate": 1526774400000,
                },
                Object {
                  "Budget": 1349838,
                  "endDate": 1546128000000,
                  "id": 1004,
                  "name": "R2-D2",
                  "startDate": 1508025600000,
                },
                Object {
                  "Budget": 884850,
                  "endDate": 1532217600000,
                  "id": 1005,
                  "name": "Star City",
                  "startDate": 1531353600000,
                },
                Object {
                  "Budget": 85313,
                  "endDate": 1500940800000,
                  "id": 1006,
                  "name": "Green Arrow",
                  "startDate": 1500768000000,
                },
                Object {
                  "Budget": 109448,
                  "endDate": 1532217600000,
                  "id": 1007,
                  "name": "Stranger Things",
                  "startDate": 1531353600000,
                },
                Object {
                  "Budget": 255442,
                  "endDate": 1469145600000,
                  "id": 1008,
                  "name": "Klingons",
                  "startDate": 1468281600000,
                },
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "main",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
